src/Rules/Generics/TemplateTypeCheck.php

```diff
@@ -79,7 +79,7 @@ public function check(
 					$templateTagName
 				))->build();
 			}
-			if ($this->typeAliasResolver->hasTypeAlias($templateTagName)) {
+			if ($this->typeAliasResolver->hasTypeAlias($templateTagName, $templateTypeScope->getClassName())) {
```

Comment:
This change should have a test - `@template LocalAlias` should report an error.

Reply:
Added in [643ffd89](https://github.com/phpstan/phpstan-src/pull/460/commits/643ffd8993cb7bbab2db1fc4c50b2a4978fa6964)